packages/react-composites/tests/browser/callwithchat/CallWithChatComposite.test.ts

```diff
@@ -70,6 +70,48 @@ test.describe('CallWithChat Composite CallWithChat Page Tests', () => {
     expect(await pages[0].screenshot()).toMatchSnapshot(`call-with-chat-gallery-screen-with-chat-pane.png`);
   });
 
+  test('Unread chat message button badge are displayed correctly for <9 messages', async ({ pages }) => {
+    // Open chat pane on page 0 and send a message
+    await pageClick(pages[0], dataUiId('call-with-chat-composite-chat-button'));
+    await waitForSelector(pages[0], dataUiId('call-with-chat-composite-chat-pane'));
+    await sendMessage(pages[0], 'Call with Chat composite is awesome!');
+    await waitForMessageDelivered(pages[0]);
+
+    // Ensure typing indicator has disappeared to prevent flakey test
+    await pages[1].bringToFront();
+    const typingIndicator = await pages[1].$(dataUiId(IDS.typingIndicator));
+    typingIndicator && (await typingIndicator.waitForElementState('hidden'));
+    await waitForSelector(pages[1], dataUiId('call-with-chat-composite-chat-button-unread-icon'));
+    expect(await pages[1].screenshot()).toMatchSnapshot(`call-with-chat-gallery-screen-with-one-unread-messages.png`);
+  });
+
+  test('Unread chat message button badge are displayed correctly for >9 messages', async ({ pages }) => {
+    // Open chat pane on page 0 and send 10 messages
+    await pageClick(pages[0], dataUiId('call-with-chat-composite-chat-button'));
+    await waitForSelector(pages[0], dataUiId('call-with-chat-composite-chat-pane'));
+
+    for (let i = 0; i < 10; i++) {
+      await sendMessage(pages[0], 'Call with Chat composite is awesome!');
+      // timeout between each messages to prevent chat throttling
+      await pages[0].waitForTimeout(100);
+    }
+
+    // Ensure typing indicator has disappeared to prevent flakey test
+    await pages[1].bringToFront();
+    const typingIndicator = await pages[1].$(dataUiId(IDS.typingIndicator));
+    typingIndicator && (await typingIndicator.waitForElementState('hidden'));
+
+    await waitForMessageSeen(pages[0]);
+    let seenMessages = await pages[0].$$('[data-ui-status="seen"]');
+    // make sure all messages are delivered and received before taking a screenshot
+    while (seenMessages.length < 10) {
```

Comment:
this does not have a timeout so could get stuck in an infinite loop if there are never `>=10` .
Instead can you use the `waitForFunction`?

waitForFunction sends a function for the headless browser page to execute, and continually sends that function (at some small interval) until it returns true or the global timeout elapses.

something like:
```tsx
waitForFunction(page, () => {
  // fun fact: everything inside this function body is executed by the headless browser page and not the playwright node process
  const seenMessages = document.querySelectorAll(`[data-ui-status="seen"]`);
  return seenMessages.length >= 10
});
```

Or you make a generic function for other tests to use:
```tsx
const waitForNSeenMessages(n: number): Promise<void> => {
  return await waitForFunction(page, (args) => {
  // fun fact: everything inside this function body is executed by the headless browser page and not the playwright node process
  const seenMessages = document.querySelectorAll(`[data-ui-status="seen"]`);
  return seenMessages.length >= args.seenMessageCountToLookFor
},
// args parameter to pass from the node process to the headless browser page
{
  seenMessageCountToLookFor: n
});
};

```